### DISCUSSION TITLE
Distribute Milvus Over Multiple Machines

### DISCUSSION THREAD

**Discussion by Neltherion** [milvus-io/milvus#22602] [Q&A and General discussion]
Hi

I wanted to know if it's possible to distribute Milvus over multiple machines. 

For example if my server was bottle-necked by CPU or RAM, I want to know if it's possible to go to another machine, start up another Milvus and then somehow coordinate the Milvus of the first machine to sync data with the second machine and theoretically speed up each vector search by 2 times (since we have two machines now and each search can take half the time and half the data on each machine)!

As I read the Milvus guides I somehow felt that the Helm K8s Milvus cluster is designed to be run on the same machine and we're only increasing the number of containers inside that one machine. I'd be glad to be corrected if I'm wrong.

Thanks

**Comment by yhmo** [comment]
Currently, Milvus has no ability to instantly sync data across multiple clusters/instances. We have backup/restore tool for backup data from a cluster and import into another cluster, but it is not instantly.

Helm can deploy milvus cluster onto different virtual machines, in the cloud, these virtual machines could be in one physical machine, but also could be on different physical machines.

**Reply by Neltherion** [reply]
Thanks

Does having multiple instances in various scenarios (virtual, cloud, physical) speed up the vector search (2 Instances = Half the search time with 1 Instance) or just increase availability (more requests can be processed instead of dropped)?

And If there is a vector search speed up with multiple instances, How is that possible with the same CPU on the same machine? Shouldn't using multiple instances that do the same thing on the CPU actually slow down the CPU versus when using one instance that fully utilizes the CPU?

And the last question, If there is speed up with more than one instance somehow, is there a hard limit on that like Amdahl's Law where having more than `n` instances wouldn't help speed up the process anymore?

Thanks Again
